packages/ckbtc/src/utils/btc.utils.ts

```diff
@@ -155,27 +155,31 @@ const parseBip173Address = ({
 
   const [witnessVersion, ...rest] = words;
 
-  if (witnessVersion !== 0) {
+  if (witnessVersion > 1) {
     throw new ParseBtcAddressUnsupportedWitnessVersionError();
   }
 
   const data = bech32.fromWords(rest);
+  let type = BtcAddressType.P2wpkhV0;
 
-  if (data.length !== 20) {
-    // Note: We throw an error for other length because the Minter canister applies such a policy and does not support P2tr and P2wsh yet.
-    // else if (witnessVersion === 1) {
-    //     type = BtcAddressType.P2tr;
-    //   } else {
-    //     type = BtcAddressType.P2wsh;
-    //   }
+  if (data.length === 32) {
+    if (witnessVersion == 0) {
```

Comment:
An opinionated feedback (😅) but, would be cool if you could avoid the use of an empty`else` and write this assignation in an immutable way.

For example:

```
const parseType = (): BtcAddressType => {
  if (data.length !== 20 && data.length !== 32) {
    throw new ParseBtcAddressBadWitnessLengthError();
  }
  
  if (data.length === 20) {
     return BtcAddressType.P2wpkhV0;
  }
  
  switch (witnessVersion) {
      case 0:
        return BtcAddressType.P2wsh;
      case 1:
        return BtcAddressType.P2tr;
      default:
        return BtcAddressType.P2wpkhV0;
    }
}

type = parseType();
```

Reply:
Good point I switched to 'switch' :)

Reply:
Thanks!

Just added a minor comment about it https://github.com/dfinity/ic-js/pull/370#discussion_r1250524849